reader.go

```diff
@@ -93,18 +121,53 @@ func (r *Reader) NumRead() int {
 
 // LastModified returns the last modified time of the remote feed.
 func (r *Reader) LastModified() (time.Time, error) {
-	lastMod, err := remoteLastModified(r.ctx, r.remote)
-	return lastMod.Time(), err
+	var lastMod timestamp
+	for _, remote := range r.remotes {
+		t, err := remoteLastModified(r.ctx, remote)
+		if err != nil {
+			return time.Time{}, err
+		}
+		if t > lastMod {
+			lastMod = t
+		}
+	}
+
+	return lastMod.Time(), nil
 }
 
 // Close closes the reader.
 func (r *Reader) Close() error {
-	var err error
-	if r.fd != nil {
-		if e := r.fd.Close(); e != nil {
-			err = e
-		}
+	if r.cur != nil {
+		return r.cur.Close()
+	}
+	return nil
```

Comment:
Don't you want to also close all the `r.remotes` here?

Or that should be up to caller? (maybe this makes sense to leave it up to caller :thinking: )